docs/contributing.md

```diff
@@ -115,6 +115,43 @@ To spawn the docs server run:
 $ mkdocs serve
 ```
 
+## Resolving Build / Travis Failures
+
+Once you've submitted your pull request, the test suite will automatically run, and the results will show up in GitHub.
+If the test suite fails, you'll want to click through to the "Details" link, and try to identify why the test suite failed.
+
+Here are some common ways the test suite can fail:
+
+### NOX_SESSION=check Job Failed
+
+This job failing means there is either a code formatting issue or type-annotation issue.
+You can look at the job output to figure out why it's failed or within a shell run:
+
+`nox -s check`
+
+It may be worth it to run `nox -s lint` to attempt auto-formatting the code
+and if that job succeeds commit the changes.
```

Comment:
Has anyone used [`pre-commit`](https://pre-commit.com/) before? It might be good option to avoid this situation, I fall for this literally every time I push 😅

Reply:
I have previously used it with the Black plugin, but does it allow running arbitrary commands like our lint task? Or maybe it has a nox plugin too… I know they wrote a comprehensive tutorial recently so probably worth checking it out to see if it fits our use case. :)

Reply:
I think it's a bit too heavy touch personally.
I'd rather we just make sure that we've got a dev setup where eg. `scripts/test` or whatever will run the tests and linting, and enforce 100% coverage, and generally ensure that you're on track to pass the CI.